### PR TITLE
Sig 2532 zoom to incident

### PR DESCRIPTION
**Pull request by aditudorache**
This PR persitsts the selections in het MapInput and MapSelect when coming back from the incident preview

### CODE REVIEW

File: src/components/MapInput/index.js [hunk 1]
```diff
@@ -115,10 +116,18 @@ const MapInput = ({ className, value, onChange, mapOptions, ...otherProps }) =>
...
-    if (!value?.geometrie?.coordinates && !value?.addressText) return;
+    if (!map) return;
+    if (!value?.location && !value?.addressText) return;
+
+    if (value?.location?.lat > 0) {
+      if (map && !isEqual(value?.location, state.location)) {
+        const currentZoom = map.getZoom();
+        map.flyTo(value?.location, currentZoom);
+      }
+    }
```
Comment (janjaap) [inline src/components/MapInput/index.js:127]: This code block isn't covered by tests

File: src/components/MapInput/index.js [hunk 2]
```diff
@@ -115,10 +116,18 @@ const MapInput = ({ className, value, onChange, mapOptions, ...otherProps }) =>
   }, [marker, location, hasLocation]);
 
   useEffect(() => {
-    if (!value?.geometrie?.coordinates && !value?.addressText) return;
+    if (!map) return;
+    if (!value?.location && !value?.addressText) return;
+
+    if (value?.location?.lat > 0) {
+      if (map && !isEqual(value?.location, state.location)) {
```
Comment (janjaap) [inline src/components/MapInput/index.js:123]: The check for `map` is redundant; this code cannot be reached whenever `map` is falsy (see line 119).

File: src/components/MapInput/index.js [hunk 3]
```diff
@@ -115,10 +116,18 @@ const MapInput = ({ className, value, onChange, mapOptions, ...otherProps }) =>
...
+        const currentZoom = map.getZoom();
+        map.flyTo(value?.location, currentZoom);
```
Comment (janjaap) [inline src/components/MapInput/index.js:125]: No need for optional chaining here; we know that `value` has a prop `location` at this point.

File: src/components/MapInput/index.js [hunk 4]
```diff
@@ -115,10 +116,18 @@ const MapInput = ({ className, value, onChange, mapOptions, ...otherProps }) =>
...
   useEffect(() => {
-    if (!value?.geometrie?.coordinates && !value?.addressText) return;
+    if (!map) return;
+    if (!value?.location && !value?.addressText) return;
+
+    if (value?.location?.lat > 0) {
+      if (!isEqual(value?.location, state.location)) {
+        const currentZoom = map.getZoom();
+        map.flyTo(value.location, currentZoom);
+      }
```
Comment (janjaap) [inline src/components/MapInput/index.js:126]: This clause still doesn't seem to be properly covered by a test case.

File: src/components/MapInput/index.js [hunk 5]
```diff
@@ -115,10 +116,18 @@ const MapInput = ({ className, value, onChange, mapOptions, ...otherProps }) =>
   }, [marker, location, hasLocation]);
 
   useEffect(() => {
-    if (!value?.geometrie?.coordinates && !value?.addressText) return;
+    if (!map) return;
+    if (!value?.location && !value?.addressText) return;
```
Comment (janjaap) [inline src/components/MapInput/index.js:120]: I don't see `value.addressText` being used in this function. Does it need to be present?

File: src/components/MapInput/index.js (hunk 4 above)
Comment (aditudorache) [reply src/components/MapInput/index.js:126]: Done

File: src/components/MapInput/index.js (hunk 5 above)
Comment (aditudorache) [reply src/components/MapInput/index.js]: removed

**Comment by aditudorache** [comment]
closed through #779

**Pull request closed without merging**